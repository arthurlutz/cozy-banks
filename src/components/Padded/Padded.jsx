import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import styles from 'components/Padded/Padded.styl'

export const Padded = ({ className, ...restProps }) => (
  <div className={cx(styles.Padded, className)} {...restProps} />
)

Padded.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
}

const Unpadded_ = ({ horizontal, vertical, className, ...restProps }) => {
  const all = horizontal === undefined && vertical === undefined
  return (
    <div
      className={cx(
        {
          [styles['Unpadded--horizontal']]: horizontal || all,
          [styles['Unpadded--vertical']]: vertical || all
        },
        className
      )}
      {...restProps}
    />
  )
}

export const Unpadded = React.memo(Unpadded_)

const MemoPadded = React.memo(Padded)
MemoPadded.Unpadded = Unpadded

export default MemoPadded
