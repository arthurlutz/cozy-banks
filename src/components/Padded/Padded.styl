@require 'settings/breakpoints'

.Padded
    padding 1.5rem 2rem

    +small-screen()
        padding 1rem

.Unpadded--vertical
    margin-top -1.5rem
    margin-bottom -1.5rem

    +small-screen()
        margin-top -1rem
        margin-bottom -1rem

.Unpadded--horizontal
    margin-left -2rem
    margin-right -2rem

    +small-screen()
        margin-left -1rem
        margin-right -1rem
