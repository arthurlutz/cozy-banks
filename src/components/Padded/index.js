export { default, Unpadded } from './Padded'
