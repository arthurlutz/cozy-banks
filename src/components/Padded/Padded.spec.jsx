import React from 'react'
import { shallow } from 'enzyme'
import { Padded } from './Padded'

describe('Padded', () => {
  it(`should display children`, () => {
    expect(shallow(<Padded>content</Padded>).getElement()).toMatchSnapshot()
  })

  it(`should extend className`, () => {
    expect(
      shallow(<Padded className="noPaddingBottom">content</Padded>).getElement()
    ).toMatchSnapshot()
  })
})
