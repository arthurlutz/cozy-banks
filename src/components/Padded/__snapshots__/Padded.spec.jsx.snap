// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Padded should display children 1`] = `
<div
  className="Padded"
>
  content
</div>
`;

exports[`Padded should extend className 1`] = `
<div
  className="Padded noPaddingBottom"
>
  content
</div>
`;
